shell/platform/linux/fl_engine.cc

```diff
@@ -59,6 +94,20 @@ static bool fl_engine_gl_present(void* user_data) {
   return result;
 }
 
+static bool fl_engine_runs_task_on_current_thread(void* user_data) {
+  FlEngine* self = static_cast<FlEngine*>(user_data);
+  return self->thread == g_thread_self();
+}
+
+static void fl_engine_post_task_callback(FlutterTask task,
+                                         uint64_t target_time,
+                                         void* user_data) {
+  FlEngine* self = static_cast<FlEngine*>(user_data);
+
+  g_idle_add_full(G_PRIORITY_HIGH_IDLE, run_task_in_idle,
```

Comment:
This call doesn't use `target_time` in any way, so I don't see how this code can be correct. Am I missing something?

Reply:
Good catch. Sorry I missed this. Clearing my review.

Reply:
Ah, I misinterpreted this as a timestamp like the event timestamp. I've now reworked this branch to have much deeper integration in the GLib main loop using this scheduled time.